### ISSUE TITLE
False positive error when matching instance against a generic type whose type arguments weren't specified

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#1517]
```python
Integer = TypeVar('Integer')
class Ndim(Generic[Integer]): ...

Axes = TypeVarTuple('Axes')
class Shape(Generic[*Axes]): ...

DataType = TypeVar('DataType')
ShapeType = TypeVar('ShapeType', Ndim, Shape)

class Array(Generic[DataType, ShapeType]): ...

Float32Array = Array[int, Shape[*Axes]]  # False Positive
Array1D = Array[DataType, Ndim[int]]  # False Positive

```

**Comment by erictraut** [comment]
This will be fixed in the next release

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.115, which was just published. It will also be in the next release of Pylance.
